src/imap/select_folder.rs

```diff
@@ -146,15 +146,21 @@ impl ImapSession {
                 },
             }
         };
+        let transport_id = self.transport_id();
+
+        // Folders should not be selected when transport_id is not assigned yet
+        // because we cannot save UID validity then.
+        debug_assert!(transport_id > 0);
```

Comment:
There are `ensure_and_debug_assert` and `logged_debug_assert` macros, `debug_assert` shouldn't be used directly normally